### PR TITLE
[cleanup] Replace SafeTickEvent with ClientTickEvent

### PR DESCRIPTION
**Pull request by huddy987**

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ object ForgeEventProcessor {
         KamiEventBus.post(event)
 
         if (mc.world != null && mc.player != null) {
-            SafeTickEvent(event.phase).also {
+            TickEvent.ClientTickEvent(event.phase).also {
                 KamiEventBus.post(it)
             }
         }
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt:40]: These lines are for posting the `SafeTickEvent`. It should be removed if you already replaced all the usages and have the event class removed.

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/AutoOffhand.kt [hunk 2]
```diff
@@ -95,8 +96,8 @@ object AutoOffhand : Module() {
             if (!transactionLog.containsValue(false)) movingTimer.reset(-175L) // If all the click packets were accepted then we reset the timer for next moving
         }
 
-        listener<SafeTickEvent>(1100) {
-            if (mc.player.isDead || !movingTimer.tick(200L, false)) return@listener // Delays 4 ticks by default
+        safeListener<TickEvent.ClientTickEvent>(1100) {
+            if (mc.player.isDead || !movingTimer.tick(200L, false)) return@safeListener // Delays 4 ticks by default
             if (!mc.player.inventory.itemStack.isEmpty) { // If player is holding an in inventory
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/AutoOffhand.kt:101]: All usages such as `mc.player`, `mc.world`, `mc.connection`, `mc.playerController` in `safeTickListener` should be replaced with the non null ones in `SafeClientEvent`

Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/AutoOffhand.kt:101]: See code in ElytraFlight for example

File: src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt (hunk 1 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt:40]: The `TickEvent.ClientTickEvent` is already posted at line 34. Null checks will be done by `safeListener`

**Review by Xiaro — CHANGES_REQUESTED** [review]

**Review by l1ving — APPROVED** [review]

**Pull request merged**